rocketpy/motors/Tank.py

```diff
@@ -0,0 +1,626 @@
+            flux is being analyzed. In general, during this time, the tank is
+            being filled or emptied.
+            If a float is given, the flux time is assumed to be between 0 and the
+            given float, in seconds. If a tuple of float is given, the flux time
+            is assumed to be between the first and second elements of the tuple.
+        gas : rocketpy.motors.Fluid
+            Gas inside the tank as a Fluid object.
+        liquid : rocketpy.motors.Fluid
+            Liquid inside the tank as a Fluid object.
+        discretize : int, optional
+            Number of points to discretize fluid inputs. If the input
+            already has a appropriate discretization, this parameter
+            must be set to None. The default is 100.
+        """
+        self.name = name
+        self.geometry = geometry
+        self.flux_time = flux_time
+        self.gas = gas
+        self.liquid = liquid
+        self.discretize = discretize
+        return None
+
+    @property
+    def flux_time(self):
+        """Returns the start and final times of the tank flux.
+
+        Returns
+        -------
+        tuple
+            Tuple containing start and final times of the tank flux.
+        """
+        return self._flux_time
+
+    @flux_time.setter
+    def flux_time(self, flux_time):
+        """Sets the start and final times of the tank flux.
+
+        Parameters
+        ----------
+        flux_time : tuple
+            Tuple containing start and final times of the tank flux.
+        """
+        self._flux_time = tuple_handler(flux_time)
+
+    @property
+    @abstractmethod
+    def mass(self):
+        """
+        Returns the total mass of liquid and gases inside the tank as a
+        function of time.
+
+        Returns
+        -------
+        Function
+            Mass of the tank as a function of time. Units in kg.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def netMassFlowRate(self):
+        """
+        Returns the net mass flow rate of the tank as a function of time.
+        Net mass flow rate is the mass flow rate exiting the tank minus the
+        mass flow rate entering the tank, including liquids and gases.
+
+        Returns
+        -------
+        Function
+            Net mass flow rate of the tank as a function of time.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def liquidVolume(self):
+        """
+        Returns the volume of the liquid as a function of time.
+
+        Returns
+        -------
+        Function
+            Volume of the liquid as a function of time.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def gasVolume(self):
+        """
+        Returns the volume of the gas as a function of time.
+
+        Returns
+        -------
+        Function
+            Volume of the gas as a function of time.
+        """
+        pass
+
```

Comment:
What do you think about this little guy?

```suggestion

@property
    def netVolume(self):
        """
        Returns the net volume of the tank as a function of time. Net
        volume is the volume of the gas plus the volume of the liquid.

        Returns
        -------
        Function
            Net volume of the tank as a function of time.
        """
        return self.gasVolume + self.liquidVolume
        
```

Reply:
Great suggestion. I have only made the implementation concrete since some tanks can have this value computed in specific ways that are more accurate. See 97c0b160fbd1f2c432e121ebff0ab1763946d78a.